src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java

```diff
@@ -177,4 +166,58 @@ public final void testListInstitution(final Vertx aVertx, final VertxTestContext
         }).onFailure(aContext::failNow);
     }
 
+    /**
+     * Tests deleting institution from db.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testDeleteInstitution(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        final Institution toDelete = TestUtils.getRandomInstitution();
+        myScheduleStoreProxy.addInstitution(toDelete).onSuccess(newID -> {
+            myScheduleStoreProxy.removeInstitution(newID).onSuccess(result -> {
+                myScheduleStoreProxy.getInstitution(newID).onFailure(details -> {
+                    final ServiceException error = (ServiceException) details;
+
+                    aContext.verify(() -> {
+                        assertEquals(Error.NOT_FOUND.ordinal(), error.failureCode());
+                        assertTrue(error.getMessage().contains(String.valueOf(newID)));
+                    }).completeNow();
+                }).onSuccess(select -> {
+                    aContext.failNow("delete failed");
+                });
+            });
+        }).onFailure(aContext::failNow);
+    }
+
+    /**
+     * Tests updating institution in db.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testUpdateInstitution(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        final int instID = 1;
+        myScheduleStoreProxy.getInstitution(instID).onSuccess(original -> {
+            final Institution modified =
+                    new Institution(original.getName(), "changing description", "changing location",
+                            original.getEmail(), original.getPhone(), original.getWebContact(), original.getWebsite());
+            myScheduleStoreProxy.updateInstitution(instID, modified).onSuccess(result -> {
+                myScheduleStoreProxy.getInstitution(instID).onSuccess(updated -> {
+                    aContext.verify(() -> {
+                        assertTrue(updated.getName().equals(original.getName()));
+                        assertTrue(updated.getDescription().equals(modified.getDescription()));
+                    }).completeNow();
+                    aContext.completeNow();
+                });
+                aContext.completeNow();
+            });
+            aContext.completeNow();
```

Comment:
These three calls to `aContext.completeNow()` need to be removed. The test should only pass if the assertions on L212 and L213 succeed; that's handled by the chained call to `completeNow` (after `verify`) on L214. And the calls to `updateInstitution` and `getInstitution` need failure handlers.

Reply:
Added fail handler, removed wrongful completes